posts/2024-03-29-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,161 @@
+
+[.img_border_light]
+image::/img/blog/configuration-drift-4.png[Setting the Repo,width=70%,align="center"]
+
+[#manually_sync]
+=== Manually Syncing your Application with the Repository
+
+Since we chose manual sync policy in the options for demonstrative purposes, we have to sync the application manually for ArgoCD to deploy the application. If you have also selected this option, on the main page of the ArgoCD console, you should see a new tile that looks like this:
+
+[.img_border_light]
+image::/img/blog/configuration-drift-5.png[Argo Dashboard,width=70%,align="center"]
+
+Click the Sync button. Then, on the window that appears, click Sync again.
+To monitor the progress of this application sync, you can click on the daytrader7 tile.
+Over the course of a few minutes, you should see resources being created and the app dashboard should look like this:
+[.img_border_light]
+image::/img/blog/configuration-drift-6.png[App deploying,width=70%,align="center"]
+
+This diagram shows the many kubernetes resources created for the application, such as the deployment, pod, and service.  We can also monitor the health of each of these resources and verify they were created properly.  
+
+
+Now that the application is deployed and all the resources are created we need to make sure it is exposed externally.  This sample sets `.spec.expose` to `true`, which automatically creates an external route on OpenShift.  If you are on Kubernetes, you need to either configure a load balancer or an Ingress controller for the application.  Find the URL via `oc get routes –n daytrader7` if on OpenShift, or `kubectl get ingress -n daytrader7` if on Kubernetes with Ingress configured.  Then paste the URL in a browser to access the Daytrader login page.
+
+And there we go! We have fully deployed our application with GitOps!
+[.img_border_light]
+image::/img/blog/configuration-drift-7.png[Daytrader Home,width=70%,align="center"]
+
+[#gitops_solve_config_drift]
+== See how GitOps can solve Configuration Drift
+
+Now that we have the app deployed let's show a scenario where configuration drift occurs and how we can easily correct it.
+Let's say a developer changes the memory settings in the configuration file for a quick test.  To represent this, change the memory request for the `deploy/daytrader7-deploy.yaml` file from 1024Mi to 2048Mi using either the Kubernetes CLI or the console itself.  In a few moments the argo console should show it is out of sync:
+[.img_border_light]
+image::/img/blog/configuration-drift-outofsync.png[Daytrader Home,width=70%,align="center"]
+
+"Out of Sync" means that some form of configuration drift occured in the application.  We can refer back to the diagram to tell exactly where it occured.  
+Since we set Auto-Sync to disable earlier in this blog, we have to manually tell ArgoCD to correct the configuration drift.  To do this, click on the `Sync` button to return the configuration to match what git says it should be.
```

Comment:
```suggestion
Since we set the sync policy to manual, we have to manually tell ArgoCD to correct the configuration drift. Click the `Sync` button to return the configuration to match what is in Git.
```